### ISSUE TITLE
Analysis server does not report analytics or crashes from non-VS Code clients

### ISSUE THREAD

**Issue by DanTup** [dart-lang/sdk#55260]
Splitting this from https://dart-review.googlesource.com/c/sdk/+/358902.

Currently, analytics and crash reports require a mapping from a client-id to a "DashTool" from unified analytics, but this is only done for VS Code:

https://github.com/dart-lang/sdk/blob/7e2679df0596972cae9854032dbf0db987832cff/pkg/analysis_server/lib/src/server/driver.dart#L232-L237

This means non-VS Code clients (including `dart analyze`, whose client ID is `dart-analyze`) are not having anything reported. While third party editors using the analysis server might be harder to map, probably first-party commands like `dart analyze` and IntelliJ (assuming it can show the notification) could still be handled here.

**Comment by DanTup** [comment]
@bwilkerson is this done? You made changes in 7f74e064b0c4cebd67762dc481e3fb91d661bba0 although it's not clear to me if there's anything left - the commit message says "any clients that can display a message", although it still looks like a hard-coded list (so for example might not include non-VS-Code LSP clients that can display messages?)

**Comment by bwilkerson** [comment]
Yes, I think there's still work to be done here. It's related to some work that @pq is probably going to be doing.
